File: library-test/src/androidTest/java/com/uphyca/creditcardedittext/ViewMatchers.java
package com.uphyca.creditcardedittext;

import android.support.test.espresso.matcher.BoundedMatcher;
import android.view.View;
import android.widget.TextView;

import org.hamcrest.Description;
import org.hamcrest.Matcher;

public class ViewMatchers {

    private ViewMatchers() {
    }

    public static Matcher<View> withSelection(final int index) {
        return new BoundedMatcher<View, TextView>(TextView.class) {
            @Override
            public void describeTo(Description description) {
                description.appendText("with selection: " + index);
            }

            @Override
            protected boolean matchesSafely(TextView item) {
                return item.getSelectionStart() == index
                        && item.getSelectionEnd() == index;
            }
        };
    }
}

File: library-test/src/androidTest/java/com/uphyca/creditcardedittext/ViewMatchers.kt
package com.uphyca.creditcardedittext

import android.view.View
import android.widget.TextView
import androidx.test.espresso.matcher.BoundedMatcher
import org.hamcrest.Description
import org.hamcrest.Matcher

object ViewMatchers {

    @JvmStatic
    fun withSelection(index: Int): Matcher<View> {
        return object : BoundedMatcher<View, TextView>(TextView::class.java) {

            override fun describeTo(description: Description) {
                description.appendText("with selection: $index")
            }

            override fun matchesSafely(item: TextView): Boolean {
                return item.selectionStart == index && item.selectionEnd == index
            }
        }
    }
}
